macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +233,403 @@ open class Button: NSButton {
 
-	/// The primary color of the button, AKA, the fill color in the primaryFilled style, and the outline in the primaryOutline style
-	@objc public var primaryColor: NSColor = defaultPrimaryColor {
-		didSet {
-			if primaryColor != oldValue {
-				needsDisplay = true
-			}
+	/// State-specific colors for foreground, background and border
+	private var contentTintColorRest: NSColor?
+	private var contentTintColorPressed: NSColor?
+	private var contentTintColorDisabled: NSColor?
+	private var backgroundColorRest: NSColor?
+	private var backgroundColorPressed: NSColor?
+	private var backgroundColorDisabled: NSColor?
+	private var borderColorRest: NSColor?
+	private var borderColorPressed: NSColor?
+	private var borderColorDisabled: NSColor?
+
+	private func updateContentTintColor() {
+		if !isEnabled {
+			contentTintColor = contentTintColorDisabled
+		} else if mouseDown {
+			contentTintColor = contentTintColorPressed
+		} else {
+			contentTintColor = contentTintColorRest
 		}
 	}
 
-	/// The primary style of the button
-	@objc public var style: ButtonStyle = .primaryFilled {
+	private func setColorValues(forStyle: ButtonStyle, accentColor: NSColor) {
+		switch forStyle {
+		case .primary:
+			contentTintColorRest = ButtonColor.neutralInverted
+			contentTintColorPressed = ButtonColor.neutralInverted?.withSystemEffect(.pressed)
+			contentTintColorDisabled = ButtonColor.brandForegroundDisabled
+			backgroundColorRest = accentColor
+			backgroundColorPressed = accentColor.withSystemEffect(.pressed)
+			backgroundColorDisabled = ButtonColor.brandBackgroundDisabled
+			borderColorRest = .clear
+			borderColorPressed = .clear
+			borderColorDisabled = .clear
+		case .secondary:
+			contentTintColorRest = .textColor
+			contentTintColorPressed = ButtonColor.neutralInverted?.withSystemEffect(.pressed)
+			contentTintColorDisabled = NSColor.textColor.withSystemEffect(.disabled)
+			backgroundColorRest = ButtonColor.neutralBackground2
+			backgroundColorPressed = accentColor.withSystemEffect(.pressed)
+			backgroundColorDisabled = ButtonColor.neutralBackground2?.withSystemEffect(.disabled)
+			borderColorRest = ButtonColor.neutralStroke2
+			borderColorPressed = .clear
+			borderColorDisabled = ButtonColor.neutralStroke2?.withSystemEffect(.disabled)
+		case .acrylic:
+			contentTintColorRest = ButtonColor.neutralForeground3
+			contentTintColorPressed = ButtonColor.neutralForeground3?.withSystemEffect(.pressed)
+			contentTintColorDisabled = ButtonColor.neutralForeground3?.withSystemEffect(.disabled)
+			backgroundColorRest = ButtonColor.neutralBackground3
+			backgroundColorPressed = ButtonColor.neutralBackground3?.withSystemEffect(.pressed)
+			backgroundColorDisabled = ButtonColor.neutralBackground3?.withSystemEffect(.disabled)
+			borderColorRest = .clear
+			borderColorPressed = .clear
+			borderColorDisabled = .clear
+		case .borderless:
+			contentTintColorRest = accentColor
+			contentTintColorPressed = accentColor.withSystemEffect(.deepPressed)
```

Comment:
Yes, .deepPressed is darker than .pressed.  This makes the color change more visible in the smaller area covered by the contentTintColor.  That's also reflected in the design docs: there they have shade 10 for brand background color when pressed, and shade 30 for brand foreground color when pressed.  I am using system effects to approximate that, rather than making the client specify a host of related colors for the accent.